File: DD/app/src/main/java/com/andruid/magic/discodruid/viewholder/PlayListViewHolder.java
package com.andruid.magic.discodruid.viewholder;

import com.andruid.magic.discodruid.databinding.LayoutPlaylistBinding;
import com.andruid.magic.mediareader.model.PlayList;

import androidx.recyclerview.widget.RecyclerView;

public class PlayListViewHolder extends RecyclerView.ViewHolder {
    private LayoutPlaylistBinding binding;

    public PlayListViewHolder(LayoutPlaylistBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(PlayList playList){
        binding.setPlaylist(playList);
        binding.executePendingBindings();
    }
}

File: DiscoDruid/app/src/main/java/com/andruid/magic/discodruid/ui/viewholder/PlaylistViewHolder.kt
package com.andruid.magic.discodruid.ui.viewholder

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.andruid.magic.discodruid.R
import com.andruid.magic.discodruid.databinding.LayoutPlaylistBinding
import com.andruid.magic.medialoader.model.Playlist

class PlaylistViewHolder(private val binding: LayoutPlaylistBinding) :
    RecyclerView.ViewHolder(binding.root) {
    companion object {
        fun from(parent: ViewGroup): PlaylistViewHolder {
            val inflater = LayoutInflater.from(parent.context)
            val binding = DataBindingUtil.inflate<LayoutPlaylistBinding>(
                inflater,
                R.layout.layout_playlist,
                parent,
                false
            )
            return PlaylistViewHolder(binding)
        }
    }

    fun bind(playlist: Playlist) {
        binding.playlist = playlist
        binding.executePendingBindings()
    }
}